app/models/manageiq/providers/lenovo/physical_infra_manager/event_catcher/stream.rb

```diff
@@ -27,12 +28,26 @@ def filter_fields
       { :operation => 'NOT', :field => 'eventClass', :value => '200' },
       { :operation => 'NOT', :field => 'eventClass', :value => '800' }
     ]
-    last_cn_event = get_last_cnn_from_events(@ems.id)
-    cn_operation = { :operation => 'GT', :field => 'cn', :value => last_cn_event.to_s }
-    fields.push(cn_operation) unless last_cn_event.nil?
+
+    unless @last_event_ems_ref.nil?
+      cn_operation = { :operation => 'GT', :field => 'cn', :value => @last_event_ems_ref.to_s }
+      fields.push(cn_operation)
+    end
     fields
   end
 
+  def raw_events(events)
```

Comment:
Oh no other way around, the method is fine I meant change the input and the output variables to have different names.

Reply:
:+1: and `parse_events` was a better method name

Reply:
@agrare  hahaha :smile:  now I changed the variables name and back the method name to parse_events.